### ISSUE TITLE
belpnex new version issue

### ISSUE THREAD

**Issue by m1371339094** [BepInEx/BepInEx#280]
I'm playing games with BepInEx and when i update my BepInEx it will change the color of the game into more grayer and slow the game down

**Comment by m1371339094** [comment]
I'm playing games with BepInEx and when i update my BepInEx it will change the color of the game (honey select 2) into more grayer and slow the game down

**Comment by ManlyMarco** [comment]
Did you test it with 5.4.13? If yes, post the output log.

**Comment by m1371339094** [comment]
So the problem is that the new version of belpnex can't cooperate with XUnity.Common and when I put XUnity.Common on the the belpn core it will lag as hell

**Comment by m1371339094** [comment]
[output_log.txt](https://github.com/BepInEx/BepInEx/files/6872716/output_log.txt)

**Comment by ghorsington** [comment]
Thanks for the log, it does indeed provide some insight into the issue. Could you also ZIP your BepInEx folder and send it over? If it's too large, could you at the very least ZIP two folders:

* `BepInEx/core`
* `BepInEx/plugins`

and send them?

**Comment by m1371339094** [comment]
[core.zip](https://github.com/BepInEx/BepInEx/files/6872737/core.zip)
[Plugins.zip](https://github.com/BepInEx/BepInEx/files/6872738/Plugins.zip)

**Comment by ghorsington** [comment]
Thanks for the ZIPs. When you updated BepInEx, did you also happen to reinstall any of your plugins? Right now it seems as if certain plugins were not automatically upgraded by BepInEx, which in turn causes some errors in plugins. 

Could you try removing `BepInEx/cache` folder and then rerun the game? Please do send the newly generated output log after you have done that.

**Comment by m1371339094** [comment]
[output_log.txt](https://github.com/BepInEx/BepInEx/files/6872744/output_log.txt)
still lagging

**Comment by ManlyMarco** [comment]
Maybe it's some plugins shimmed by a very old version of bepin that fail? @m1371339094 if you have a BepInEx_Shim_Backup folder then can you try to copy its contents into your current bepinex folder and replace existing files? Make sure to backup your current folder before that of course.

**Comment by ghorsington** [comment]
It also look like you're missing HS2API altogether which causes a error span:

```
FileNotFoundException: Could not load file or assembly 'HS2API, Version=1.12.3.0, Culture=neutral, PublicKeyToken=null' or one of its dependencies.
```

You indeed do not have that file in your plugins folder you sent. Have you ever had `HS2API.dll` in the first place?

**Comment by m1371339094** [comment]
So the problem will solve if i remove XUnity.Common which is XUnity.AutoTranslator

**Comment by m1371339094** [comment]
i download the latest version of hs2api and still lagging

**Comment by m1371339094** [comment]
but when i remove XUnity.Common.dll it will not translate

**Comment by m1371339094** [comment]
it's very associate with ShaneFreak case

**Comment by m1371339094** [comment]
[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
UnityEngine.Rendering.PostProcessing.PostProcessManager+<>c.<ReloadBaseTypes>b__12_0 (System.Type t) (at <784cbda9c5b64955b9c3ba00991a435c>:0)
System.Linq.Enumerable+WhereEnumerableIterator`1[TSource].MoveNext () (at <fbb5ed17eb6e46c680000f8910ebb50c>:0)
UnityEngine.Rendering.PostProcessing.PostProcessManager.ReloadBaseTypes () (at <784cbda9c5b64955b9c3ba00991a435c>:0)
UnityEngine.Rendering.PostProcessing.PostProcessManager..ctor () (at <784cbda9c5b64955b9c3ba00991a435c>:0)
UnityEngine.Rendering.PostProcessing.PostProcessManager.get_instance () (at <784cbda9c5b64955b9c3ba00991a435c>:0)
UnityEngine.Rendering.PostProcessing.PostProcessVolume.Update () (at <784cbda9c5b64955b9c3ba00991a435c>:0)

**Comment by ghorsington** [comment]
I got myself a basic HS2 (version 1.2.3) setup to test and it does work just fine with BepInEx 5.4.13 and all generally used plugins. 
I also failed to reproduce any perceivable lag on my laptop (16 GB RAM, GeForce GTX 1050 Mobile with 2GB VRAM).

When trying to using the plugins folder you provided, it did indeed fail with a multitude of errors and mentions of missing plugins.

I would consider cleaning up your install. Here is a ZIP of my working BepInEx 5.4.13 folder, my plugins and scripts:

https://www.swisstransfer.com/d/802de8e4-0369-406b-8d2f-4d8584fe6da1

I suggest backing up your relevant folders and files and then replacing them with the ones in the ZIP, **not overwrite what you have with the ZIP**.
